client/pubsub.go

```diff
@@ -3,61 +3,80 @@ package client
 import (
 	"context"
 	"encoding/json"
+	"log"
 
 	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
 	"github.com/pkg/errors"
 )
 
+// PublishEventOption is the type for the functional option.
+type PublishEventOption func(*pb.PublishEventRequest)
+
 // PublishEvent publishes data onto specific pubsub topic.
-func (c *GRPCClient) PublishEvent(ctx context.Context, pubsubName, topicName string, data []byte) error {
+func (c *GRPCClient) PublishEvent(ctx context.Context, pubsubName, topicName string, data interface{}, opts ...PublishEventOption) error {
 	if pubsubName == "" {
 		return errors.New("pubsubName name required")
 	}
 	if topicName == "" {
 		return errors.New("topic name required")
 	}
 
-	envelop := &pb.PublishEventRequest{
+	envelope := &pb.PublishEventRequest{
```

Comment:
Nit: rename `envelope` to `request` because technically it isn't a Cloud Event envelope.

Reply:
I will rename it. I just used the old name (less the typo), but agree with you